plugins/generator-1.18.2/forge-1.18.2/templates/gui/gui_container.java.ftl

```diff
@@ -144,9 +145,14 @@ public class ${name}Menu extends AbstractContainerMenu implements Supplier<Map<I
 						}
         	        <#elseif component.getClass().getSimpleName() == "InputSlot">
 						<#if component.inputLimit.toString()?has_content>
-        	             @Override public boolean mayPlace(ItemStack stack) {
-							 return (${mappedMCItemToItem(component.inputLimit)} == stack.getItem());
-						 }
+                            @Override public boolean mayPlace(ItemStack itemstack) {
+                                <#if component.inputLimit.getUnmappedValue().startsWith("TAG:")>
+                                    <#assign tag = "\"" + component.inputLimit.getUnmappedValue().replace("TAG:", "") + "\"">
+                                    return itemstack.is(TagKey.create(Registry.ITEM_REGISTRY, ${toResourceLocation(tag)}));

```

Comment:
You can simplify the code as `ItemTags.create(new ResourceLocation(${tag}))` (similar to the "Is item tagged as" procedure block). You don't need to call `toResourceLocation` because the name is already validated by the item selector
(Same for 1.19)